### PR TITLE
feat!: add rate limit module

### LINKED ISSUE #3000: IBC Rate Limit integration
# Problem

The IBC Rate Limit module needs to be integrated and e2e tests need to be added to test the integration. See https://github.com/cosmos/gaia/issues/1945 for more context. 

# Closing criteria

The [rate limit module](https://github.com/Stride-Labs/ibc-rate-limiting) is added to Gaia.

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: https://github.com/cosmos/gaia/issues/3000

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API, client, or state breaking change (i.e., requires minor or major version bump)
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [x] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry in `.changelog` (for details, see [contributing guidelines](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#changelog))
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

**Review by MSalopek — COMMENTED** [review]
The module was not added in the upgrade constants.

It is not wired correctly without it.

Check here:
* https://github.com/cosmos/gaia/blob/main/app/upgrades/v15/constants.go#L21
* https://github.com/cosmos/gaia/pull/3001/files#diff-2a787dd0bd1cd4f70265cb9dcf67f74b2634214ae684744b4069fd6d32a310c1R19

File: tests/e2e/e2e_rate_limit_test.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+	proposalRemoveRateLimitAtomFilename = "proposal_remove_rate_limit_atom.json"
+)
+
+func (s *IntegrationTestSuite) writeAddRateLimitAtomProposal(c *chain) {
+	template := `
+	{
+		"messages": [
+		 {
+		  "@type": "/ratelimit.v1.MsgAddRateLimit",
+		  "authority": "cosmos10d07y265gmmuvt4z0w9aw880jnsr700j6zn9kn",
```
Comment (sainoe) [inline tests/e2e/e2e_rate_limit_test.go]: I suppose this address is the gov module account? If this is the case,  can please put it into a constant?

**Review by MSalopek — COMMENTED** [review]
You need to bump the timeout on e2e tests to 30-35 minutes.

Upgrade test is expected to fail since v16 upgrade handlers don't work correctly - that's for another PR.

**Comment by mpoke** [comment]
> You need to bump the timeout on e2e tests to 30-35 minutes.

It's already 45m, i.e., 
```
  test-e2e:
    runs-on: ubuntu-latest
    timeout-minutes: 45
```

**Review by MSalopek — APPROVED** [review]
LGTM!

Thanks for adding the tests.

**Review by sainoe — APPROVED** [review]

**Pull request merged**